### PR TITLE
Make sure that we do not use unsupported features by NanoMips

### PR DESCRIPTION
**Pull request by djtodoro**

### CODE REVIEW

File: clang/lib/Basic/Targets/Mips.h [hunk 1]
```diff
@@ -67,17 +67,18 @@ class LLVM_LIBRARY_VISIBILITY MipsTargetInfo : public TargetInfo {
         IsSingleFloat(false), IsNoABICalls(false), CanUseBSDABICalls(false),
         FloatABI(HardFloat), DspRev(NoDSP), HasMSA(false), DisableMadd4(false),
         UseIndirectJumpHazard(false), FPMode(FPXX) {
-
     TheCXXABI.set(TargetCXXABI::GenericMIPS);
 
-    if (Triple.isMIPS32())
+    if (Triple.isMIPS32()){
```
Comment (cme) [inline clang/lib/Basic/Targets/Mips.h]: Should be a space between `)` and `{`

File: llvm/lib/Target/Mips/MipsTargetObjectFile.cpp [hunk 2]
```diff
@@ -100,6 +100,9 @@ IsGlobalInSmallSectionImpl(const GlobalObject *GO,
   if (!Subtarget.useSmallSection())
     return false;
 
+  assert(!Subtarget.hasNanoMips() &&
+         "NanoMips do not support small data section");
```
Comment (cme) [inline llvm/lib/Target/Mips/MipsTargetObjectFile.cpp]: "do" -> "does"

**Review by cme — COMMENTED** [review]
Great. Two small stylistic comments. Thanks.

File: clang/include/clang/Basic/DiagnosticCommonKinds.td [hunk 3]
```diff
@@ -306,6 +306,10 @@ def err_opt_not_valid_without_opt : Error<
   "option '%0' cannot be specified without '%1'">;
 def err_opt_not_valid_on_target : Error<
   "option '%0' cannot be specified on this target">;
+def err_feature_not_valid_on_target : Error<
+  "feature '%0' cannot be specified on this target">;
+def err_mips_legacy_opt_not_valid_on_target : Error<
```
Comment (cme) [inline clang/include/clang/Basic/DiagnosticCommonKinds.td]: Is "legacy" appropriate here? To NanoMips the flags are legacy, but from MIPS point of view they're still currently useful flags, and MIPS has more users than NanoMips. Would it be better to use the same message for 'not valid on target'?

Comment (djtodoro) [reply clang/include/clang/Basic/DiagnosticCommonKinds.td]: Eh, it makes totally sense to me. Thanks for pointing that out!

**Review by cme — APPROVED** [review]
lgtm!

**Pull request merged**